worker-markup-core/src/test/java/com/github/cafdataprocessing/worker/markup/core/MarkupDocumentEngineTest.java

```diff
@@ -66,60 +67,60 @@ public void fieldOrderingIrrelevantForHashTest() throws WorkerException, Configu
-        Assert.assertTrue("COMPARISON_HASH field should be set on the first resul from the markup worker entry point to markupDocument.",
-                firstMarkupWorkerComparisonHashOptional.isPresent());
-        NameValuePair firstMarkupWorkerComparisonHash = firstMarkupWorkerComparisonHashOptional.get();
-
-        Assert.assertEquals("COMPARISON_HASH from the Document Worker call should be the same as when called via MarkupWorkerResult entrypoint.",
-                documentComparisonHash, firstMarkupWorkerComparisonHash.value);
-
-        //repeat the markupDocument call with a new MarkupWorkerResult and verify hash returned is the same
-        Multimap<String, ReferencedData> secondSourceData = buildSourceDataForHashOrderingTest();
-
-        MarkupWorkerTask secondMarkupWorkerTask = createMarkupWorkerTask(
-                secondSourceData,
-                hashConfigurations,
-                outputFields,
-                isEmail
-        );
-        MarkupWorkerResult secondMarkupWorkerResult =
-                markupDocumentEngine.markupDocument(secondMarkupWorkerTask, dataStore, codec, config, emailSplitter);
-        Assert.assertNotNull("Should have got a markup result.", firstMarkupWorkerResult);
-        Optional<NameValuePair> secondMarkupWorkerComparisonHashOptional = secondMarkupWorkerResult.fieldList.stream()
-                .filter(fi -> fi.name.equals("COMPARISON_HASH")).findFirst();
-        Assert.assertTrue("COMPARISON_HASH field should be set on the second result from the markup worker entry point to markupDocument.",
-                secondMarkupWorkerComparisonHashOptional.isPresent());
-        NameValuePair secondMarkupWorkerComparisonHash = secondMarkupWorkerComparisonHashOptional.get();
-
-        Assert.assertEquals("COMPARISON_HASH from the Document Worker call should be the same as when called via MarkupWorkerResult entrypoint the second time.",
-                documentComparisonHash, secondMarkupWorkerComparisonHash.value);
-
-        Assert.assertEquals("COMPARISON_HASH from the Document Worker call should be the same as when called via MarkupWorkerResult entrypoint the first time.",
-                firstMarkupWorkerComparisonHash.value, secondMarkupWorkerComparisonHash.value);
+        try {
+            markupDocumentEngine.markupDocument(documentToTest, hashConfigurations, outputFields, isEmail, emailSplitter);
+            com.hpe.caf.worker.document.model.Field documentComparisonHashField = documentToTest.getField("COMPARISON_HASH");
+            Assert.assertTrue("COMPARISON_HASH field on worker-document returned should have values.",
+                    documentComparisonHashField.hasValues());
+            String documentComparisonHash = documentComparisonHashField.getStringValues().get(0);
+
+            Codec codec = new JsonCodec();
```

Comment:
JsonCodec - I think should should also be changed to use other library and not have even a test dependency on the JsonCodec

Reply:
Needs to be a Codec as this is passed on the call to markupDocument whose method signature requires a codec. A test only codec implementation seems like it wouldn't offer any value here as this is just a test in markup-core.